### PR TITLE
add docs for datapoints

### PR DESCRIPTION
**Pull request by pmeier**
I only added the skeleton and added proper docstrings for `BoundingBox` and `BoundingBoxFormat`. The other ones have just copy-pasted docstrings since sphinx wouldn't build without.

Given that we are going to copy paste for all other datapoints, this can already be reviewed. I'll send proper docstrings for everything when we have reached consensus on what it should look like.

### CODE REVIEW

File: torchvision/datapoints/_bounding_box.py [hunk 1]
```diff
@@ -10,12 +10,35 @@
 
 
 class BoundingBoxFormat(StrEnum):
+    """[BETA] Coordinate format of a bounding box.
```
Comment (NicolasHug) [inline torchvision/datapoints/_bounding_box.py:13]: I thought we wanted to remove the `StrEnum` part? Or do I misremember?

File: torchvision/datapoints/_bounding_box.py [hunk 2]
```diff
@@ -10,12 +10,35 @@
...
     XYXY = StrEnum.auto()
     XYWH = StrEnum.auto()
     CXCYWH = StrEnum.auto()
 
 
 class BoundingBox(Datapoint):
+    """[BETA] :class:`torch.Tensor` subclass for bounding boxes.
+
+    Args:
+        data: Any data that can be turned into a tensor with :func:`torch.as_tensor`.
```
Comment (NicolasHug) [inline torchvision/datapoints/_bounding_box.py:31]: No strong opinion, numpy and scikit-learn use `array-like` a lot I believe

```suggestion
        data (tensor-like): Any data that can be turned into a tensor with :func:`torch.as_tensor`.
```

File: torchvision/datapoints/_bounding_box.py (hunk 1 above)
Comment (pmeier) [reply torchvision/datapoints/_bounding_box.py:13]: Remove what. The class or the functionality to instantiate from a str? IIRC yes for the former and no for the latter. If that is right, should we do it in a follow-up PR?

Comment (NicolasHug) [reply torchvision/datapoints/_bounding_box.py:13]: yeah just the inheritance from `StrEnum` - I love using strings lol. And yes, this is for another PR

**Comment by pmeier** [comment]
@NicolasHug sphinx fails to add cross links for `int` and `type` in `dtype` (WTF?), leaving us with this mess:

Screenshot from 2023-02-24 10-10-05

Any idea how to get around that?

**Comment by NicolasHug** [comment]
for `int` just use `int` instead of `ints`, for dtype I'll add it as a todo later

**Comment by pmeier** [comment]
> for `int` just use `int` instead of `ints`,

Doesn't work :disappointed: 

Screenshot from 2023-02-24 11-27-47

**Comment by pmeier** [comment]
As discussed offline, this is a more general issue and not related to this PR.

**Review by NicolasHug — APPROVED** [review]
Thanks Philip

**Pull request merged**